packages/mask/src/components/InjectedComponents/ProfileTabContent.tsx

```diff
@@ -54,16 +63,55 @@ export function ProfileTabContent(props: ProfileTabContentProps) {
         useSocialAddressListAll(identity)
     const { value: personaList = EMPTY_LIST, loading: loadingPersonaList } = useNextIDBoundByPlatform(
         activatedSocialNetworkUI.configuration.nextIDConfig?.platform as NextIDPlatform | undefined,
-        identity.identifier?.userId,
+        identity.identifier?.userId?.toLowerCase(),
+    )
+
+    const currentPersonaBinding = first(
+        personaList.sort((a, b) => sortPersonaBindings(a, b, identity.identifier?.userId?.toLowerCase() ?? '')),
     )
 
+    const isOwn = currentIdentity?.identifier?.userId?.toLowerCase() === identity?.identifier?.userId?.toLowerCase()
+
+    const personaPublicKey = isOwn
+        ? currentConnectedPersona?.identifier?.publicKeyAsHex
+        : currentPersonaBinding?.persona
+
+    const { value: personaProof, retry: retryProof } = useAsyncRetry(async () => {
+        if (!personaPublicKey) return
+        return NextIDProof.queryExistedBindingByPersona(personaPublicKey)
+    }, [personaPublicKey])
+
+    useEffect(() => {
+        return MaskMessages.events.ownProofChanged.on(() => {
+            retryProof()
+        })
+    }, [retryProof])
+
+    const wallets = personaProof?.proofs?.filter((proof) => proof?.platform === NextIDPlatform.Ethereum)
+
+    const addressList = useMemo(() => {
+        if (wallets?.length === 0 || !wallets || (!isOwn && socialAddressList?.length)) {
+            return socialAddressList
+        }
+        const addresses = wallets.map((proof) => {
+            return {
+                networkSupporterPluginID: NetworkPluginID.PLUGIN_EVM,
+                type: SocialAddressType.KV,
+                label: proof?.identity,
+                address: proof?.identity,
+            }
+        })
+        return [...socialAddressList, ...addresses]
+    }, [socialAddressList, wallets?.length, isOwn])
```

Comment:
Not just `wallets?.length`.
```suggestion
    const addressList = useMemo(() => {
        if (!wallets?.length || (!isOwn && socialAddressList?.length)) {
            return socialAddressList
        }
        const addresses = wallets.map((proof) => {
            return {
                networkSupporterPluginID: NetworkPluginID.PLUGIN_EVM,
                type: SocialAddressType.KV,
                label: proof?.identity,
                address: proof?.identity,
            }
        })
        return [...socialAddressList, ...addresses]
    }, [socialAddressList, wallets, isOwn])
```

Reply:
Since `wallets` is added as dependency of `addressList`, we should memo it as well.
```diff
diff --git a/packages/mask/src/components/InjectedComponents/ProfileTabContent.tsx b/packages/mask/src/components/InjectedComponents/ProfileTabContent.tsx
index 57792c1c3..05e879336 100644
--- a/packages/mask/src/components/InjectedComponents/ProfileTabContent.tsx
+++ b/packages/mask/src/components/InjectedComponents/ProfileTabContent.tsx
@@ -87,7 +87,10 @@ export function ProfileTabContent(props: ProfileTabContentProps) {
         })
     }, [retryProof])

-    const wallets = personaProof?.proofs?.filter((proof) => proof?.platform === NextIDPlatform.Ethereum)
+    const wallets = useMemo(
+        () => personaProof?.proofs?.filter((proof) => proof?.platform === NextIDPlatform.Ethereum),
+        [personaProof?.proofs],
+    )

     const addressList = useMemo(() => {
         if (wallets?.length === 0 || !wallets || (!isOwn && socialAddressList?.length)) {  
```